source/dub/generators/generator.d

```diff
@@ -89,18 +89,20 @@ class ProjectGenerator
 
 		if (!settings.config.length) settings.config = m_project.getDefaultConfiguration(settings.platform);
 
-		TargetInfo[string] targets;
 		string[string] configs = m_project.getPackageConfigs(settings.platform, settings.config);
+		TargetInfo[string] targets;
 
 		foreach (pack; m_project.getTopologicalPackageList(true, null, configs)) {
-			BuildSettings buildsettings;
-			buildsettings.processVars(m_project, pack, pack.getBuildSettings(settings.platform, configs[pack.name]), true);
-			prepareGeneration(pack, m_project, settings, buildsettings);
+			BuildSettings buildSettings;
+			auto config = configs[pack.name];
+			buildSettings.processVars(m_project, pack, pack.getBuildSettings(settings.platform, config), true);
+			targets[pack.name] = TargetInfo(pack, [pack], config, buildSettings);
```

Comment:
If this is going to stay like this, there should be a comment here stating that all dependencies are assumed to be a target here and the non-targets are filtered out at the end of `configureTargets`. It was quite surprising to read this and the place where the map is cleaned up is not really obvious, considering the complexity of the algorithm.

Reply:
Well, what you call non-targets are targets with TargetType `"sourceLibrary"` or `"none"`, so they are still processed, they just don't have any outputs.

Reply:
I've renamed `configureTargets -> confgurePackages` and added a note to https://github.com/dlang/dub/pull/1263/files#diff-92bd5c95e7074c6c3b0d83c3b5e15356R164. I think it's the right place to modify the target list in this function, e.g. consider a feature for package configs with multiple build targets.